arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -3488,28 +3488,36 @@ Result RestReplicationHandler::createBlockingTransaction(
 
   std::string vn = _vocbase.name();
   try {
-    std::function<void(void)> f = [=]() {
-      try {
-        // Code does not matter, read only access, so we can roll back.
-        transaction::Manager* mgr = transaction::ManagerFeature::manager();
-        if (mgr) {
-          mgr->abortManagedTrx(id, vn);
+    if (!serverId.empty()) {
+      std::string comment = std::string("SynchronizeShard from ") + serverId +
```

Comment:
If we declare it befor reclaring `f` below - we will force `message` to be copied into capture. Is that intended, given `f` does not use `message`?

Reply:
Ah no, seems that I`m wrong. Only used is copied into capture.Pls, ignore this

Reply:
I think `=` does not do unnecessary copies. According ot https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture it only copies *used* variables:

>      = (implicitly capture the used automatic variables by copy).